### PR TITLE
mpi_cores

### PR DESCRIPTION
**Pull request by Ahmed-Aslam**
This fix allows MPI environment for running simulation on multiple cores.

### CODE REVIEW

File: pyiron_continuum/damask/damaskjob.py [hunk 1]
```diff
@@ -20,7 +20,8 @@ def __init__(self, project, job_name):
         self.input['grid'] = np.array([16,16,16])
         self.input['size'] = np.array([1.0,1.0,1.0])
         self.input['grains'] = 20
-        self.executable = "DAMASK_grid -l tensionX.load -g damask.vtr"
+        self.input['cores'] = 4
+        self.executable = "mpiexec -np "+str(self.input['cores'])+" DAMASK_grid -l tensionX.load -g damask.vtr"
```
Comment (jan-janssen) [inline pyiron_continuum/damask/damaskjob.py:24]:
```suggestion
        self._executable_activate()
```

Comment (jan-janssen) [reply pyiron_continuum/damask/damaskjob.py:24]: The executable activate script searches for shell scripts to link to the executable.

Comment (niklassiemer) [reply pyiron_continuum/damask/damaskjob.py:24]: Shouldn't we add DAMASK to the resources folder and 'just' use our normal setup for executables? I opened a PR to do so [here](https://github.com/pyiron/pyiron-resources/pull/41). Maybe, @Ahmed-Aslam could you add the version number to the executable name like we usually do that?

**Comment by muh-hassani** [comment]
@Ahmed-Aslam, if you apply the changes suggested by Jan and Niklas, we can merge this.

**Comment by Ahmed-Aslam** [comment]
Recent wrapper changes are inline with [https://github.com/pyiron/pyiron_continuum/blob/mpirun_cores/notebooks/Damask_first_steps.ipynb] with basic tutorial in this notebook.

File: pyiron_continuum/damask/damaskjob.py [hunk 2]
```diff
@@ -1,66 +1,166 @@
...
-        self.input['C11'] = 0
-        self.input['C12'] = 0
-        self.input['C44'] = 0
-        self.input['grid'] = np.array([16,16,16])
-        self.input['size'] = np.array([1.0,1.0,1.0])
-        self.input['grains'] = 20
-        self.input['cores'] = 4
-        self.executable = "mpiexec -np "+str(self.input['cores'])+" DAMASK_grid -l tensionX.load -g damask.vtr"
+        self.input = InputList()
+        self.executable = self.executable_activate
```
Comment (muh-hassani) [inline pyiron_continuum/damask/damaskjob.py]: @Ahmed-Aslam, why did not you just use: `self._executable_activate()`. When I checked other job like [lammpsjob](https://github.com/pyiron/pyiron_atomistics/blob/c5df5e87745d7b575463f7b2a0b588e18007dc40/pyiron_atomistics/lammps/lammps.py#L37), it should be sufficient. As mentioned by Jan, by this, pyiron looks for the executable in the resource directory.

**Review by muh-hassani — CHANGES_REQUESTED** [review]

File: pyiron_continuum/damask/damaskjob.py (hunk 2 above)
Comment (Ahmed-Aslam) [reply pyiron_continuum/damask/damaskjob.py]: I thought this was just required to search script job that should be link to executable. In the recent change it was done, but not in the same way as with other pyiron jobs, I'll update it.

**Pull request closed without merging**